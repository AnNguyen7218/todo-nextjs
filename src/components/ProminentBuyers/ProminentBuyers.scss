@import 'config.gutters';
@import 'config.colors';
@import 'config.media';

.container {
  padding-top: gutter(48);
  padding-bottom: gutter(48);
  background-color: $light-gray;

  @include break(xl) {
    padding-top: gutter(96);
    padding-bottom: gutter(96);
  }
}

.header-container {
  margin-bottom: gutter(40);

  @include break(xl) {
    margin-bottom: gutter(48);
  }
}

.slider-container {
  padding: 0;

  @include break(xl) {
    padding-right: 4rem;
    padding-left: 4rem;
  }
}

.img-container {
  display: flex;
  align-items: center;
  height: 6rem;
  margin-right: gutter(48);

  @include break(xl) {
    margin-right: gutter(80);
  }

  &-ifc {
    width: 25rem;

    @include break(lg) {
      width: 27.3rem;
    }
  }
  &-aia {
    width: 12.3rem;

    @include break(lg) {
      width: 13.3rem;
    }
  }
  &-sf {
    width: 19.8rem;

    @include break(lg) {
      width: 21.2rem;
    }
  }
  &-rb {
    width: 7.9rem;

    @include break(lg) {
      width: 8.4rem;
    }
  }
}

.img {
  width: 100%;
}
