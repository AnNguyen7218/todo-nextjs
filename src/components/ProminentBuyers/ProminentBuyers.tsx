import React from 'react';
import cx from 'classnames';
import { PROMINENT_BUYERS } from 'constants/sellerLandingPage';
import { Container, InactiveSlider } from 'components';
import css from './ProminentBuyers.scss';

const ProminentBuyers = () => {
  return (
    <div className={css['container']}>
      <Container className={css['header-container']}>
        <h2>Trusted by 5000+ businesses</h2>
      </Container>
      <Container className={css['slider-container']}>
        <InactiveSlider>
          {PROMINENT_BUYERS.map((buyer, index) => {
            const { icon: Icon, name } = buyer;
            return (
              <div
                key={index}
                className={cx(
                  css['img-container'],
                  css[`img-container-${name}`]
                )}
              >
                <Icon className={css['img']} />
              </div>
            );
          })}
        </InactiveSlider>
      </Container>
    </div>
  );
};

export default ProminentBuyers;
