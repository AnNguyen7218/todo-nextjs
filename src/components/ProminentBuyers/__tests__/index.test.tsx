import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ProminentBuyers from '../ProminentBuyers';

afterEach(cleanup);

describe('Slider Component', () => {
  test('Slider should match snapshot', () => {
    const { container } = render(<ProminentBuyers />);
    expect(container).toMatchSnapshot();
  });
});
