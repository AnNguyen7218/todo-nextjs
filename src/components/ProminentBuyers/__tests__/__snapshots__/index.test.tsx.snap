// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Component Slider should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container header-container"
    >
      <h2>
        Trusted by 5000+ businesses
      </h2>
    </div>
    <div
      class="container slider-container"
    >
      <div
        class="swiper-container swiper-container-initialized swiper-container-horizontal"
      >
        <div
          class="swiper-wrapper"
          style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide img-container img-container-ifc swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <ic-ifc.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-aia swiper-slide-duplicate"
            data-swiper-slide-index="1"
          >
            <ic-aia.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-sf swiper-slide-duplicate"
            data-swiper-slide-index="2"
          >
            <ic-sf.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-rb swiper-slide-duplicate"
            data-swiper-slide-index="3"
          >
            <ic-rb.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-ifc"
            data-swiper-slide-index="0"
          >
            <ic-ifc.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-aia"
            data-swiper-slide-index="1"
          >
            <ic-aia.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-sf"
            data-swiper-slide-index="2"
          >
            <ic-sf.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-rb"
            data-swiper-slide-index="3"
          >
            <ic-rb.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-ifc swiper-slide-duplicate"
            data-swiper-slide-index="0"
          >
            <ic-ifc.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-aia swiper-slide-duplicate"
            data-swiper-slide-index="1"
          >
            <ic-aia.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-sf swiper-slide-duplicate"
            data-swiper-slide-index="2"
          >
            <ic-sf.svg
              classname="img"
            />
          </div>
          <div
            class="swiper-slide img-container img-container-rb swiper-slide-duplicate"
            data-swiper-slide-index="3"
          >
            <ic-rb.svg
              classname="img"
            />
          </div>
        </div>
        <span
          aria-atomic="true"
          aria-live="assertive"
          class="swiper-notification"
        />
      </div>
    </div>
  </div>
</div>
`;
